src/Razor/src/Microsoft.AspNetCore.Razor.Language/Extensions/DesignTimeDirectiveTargetExtension.cs

```diff
@@ -44,6 +44,13 @@ private void WriteDesignTimeDirectiveToken(CodeRenderingContext context, Directi
                 return;
             }
 
+            if (tokenKind == DirectiveTokenKind.Attribute)
+            {
+                // We don't need to do anything special here.
+                // We let the Roslyn take care of providing syntax errors for C# attributes.
```

Comment:
This means you get 0 C# completion for attributes that don't make it into the generated C#. In practice we don't have to worry about this because our `@attribute` directive always gets represented in codegen; however, given that you added this to the extensible directive system anyone else extending is not guaranteed to have proper DesignTime bits generated for them.

Can you think of any way to include extensible directive attributes at this layer to ensure C# mappings are generated? If not I don't want to rat hole because in practice almost no one extends this portion of the Razor system; however, if it's easily addable It'd be worth doing.

Reply:
I can't think of a simple way of doing this **correctly**. Because if I create a dummy class and put it on top of this, it might show have usage errors depending on the `AttributeUsage` of that attribute.

Reply:
Alright, fair enough.